src/Lykke.SettingsReader/ReloadingManager/Configuration/IReloadingManagerWithConfiguration.cs

```diff
@@ -0,0 +1,16 @@
+﻿using Microsoft.Extensions.Configuration;
+
+namespace Lykke.SettingsReader
+{
+    /// <summary>
+    /// Interface for IReloadingManager that provides Microsoft.Extensions.Configuration.IConfiguration functionality
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public interface IReloadingManagerWithConfiguration<T> : IReloadingManager<T>
+    {
+        /// <summary>
+        /// Property that contains Microsoft.Extensions.Configuration.IConfiguration implementation for IReloadingManager
+        /// </summary>
+        IConfiguration SettingsConfiguration { get; }
```

Comment:
Will be nice to mark all public properties, method return values and method parameters, which types are reference-types, with [NotNull] or [CanBeNull]

Reply:
I decided not to add new dependency to JetBrains nuget for this